syntax = "proto3";

package SGBlocks;

message BlockHashData 
{
    bytes hash = 1;// 
}
message BlockHeaderData 
{
    bytes parent_hash = 1;// 
    uint64 block_number = 2;// 
    bytes state_root = 3; //
    bytes extrinsics_root = 4; //
    bytes digest = 5//
}
message BlockPayloadData
{
    DAGStruct dag_struct = 1;// 
    uint64 token_id = 2; // 
    bytes encrypted_amount = 3; // 
    bytes dest_addr = 4; // 
}

